src/eth1spec/spec.py

```diff
@@ -321,6 +329,56 @@ def signing_hash(tx: Transaction) -> Hash32:
     )
 
 
+def header_hash(header: Header) -> Hash32:
+    return crypto.keccak256(
+        rlp.encode(
+            (
+                header.parent_hash,
+                header.ommers_hash,
+                header.coinbase,
+                header.state_root,
+                header.transactions_root,
+                header.receipt_root,
+                header.bloom,
+                header.difficulty,
+                header.number,
+                header.gas_limit,
+                header.gas_used,
+                header.timestamp,
+                header.extra_data,
+                header.mix_digest,
+                header.nonce,
+            )
+        )
+    )
+
+
+def get_block_header_by_hash(hash, chain):
+    for block in chain.blocks:
+        if header_hash(block.header) == hash:
+            return block.header
+
+
+def validate_gas_limit(gas_limit: int, parent_gas_limit: int) -> bool:
+    if gas_limit >= parent_gas_limit + parent_gas_limit // GAS_LIMIT_ADJUSTMENT_FACTOR:
+        return False
+    if gas_limit <= parent_gas_limit - parent_gas_limit // GAS_LIMIT_ADJUSTMENT_FACTOR:
+        return False
+    if gas_limit < GAS_LIMIT_MINIMUM:
+        return False
+
+    return True
+
+def calculate_block_difficulty(header: Header, parent_header: Header) -> bool:
+    if parent_header.number == 0:
+        return 131072
+    elif header.timestamp < parent_header.timestamp:
+        return parent_header.difficulty + parent_header.difficulty // 2048
+    else:
+        return map(131072, parent_header.difficulty - parent_header.difficulty // 2048)
```

Comment:
```suggestion
        return max(131072, parent_header.difficulty - parent_header.difficulty // 2048)
```

Reply:
Thanks for catching this. I had this fixed on my local machine.